Mathlib/Algebra/BigOperators/Field.lean

```diff
@@ -20,3 +21,33 @@ lemma Multiset.sum_map_div (s : Multiset ι) (f : ι → K) (a : K) :
 
 lemma Finset.sum_div (s : Finset ι) (f : ι → K) (a : K) :
     (∑ i ∈ s, f i) / a = ∑ i ∈ s, f i / a := by simp only [div_eq_mul_inv, sum_mul]
+
+-- TODO: Move these once `Finset.dens` doesn't depend on `Field` anymore
+namespace Finset
+variable {α β : Type*} [Fintype β]
+
+@[simp]
+lemma dens_disjiUnion (s : Finset α) (t : α → Finset β) (h) :
+    (s.disjiUnion t h).dens = ∑ a ∈ s, (t a).dens := by simp [dens, sum_div]
+
+variable {s : Finset α} {t : α → Finset β}
+
+lemma dens_biUnion [DecidableEq β] (h : ∀ x ∈ s, ∀ y ∈ s, x ≠ y → Disjoint (t x) (t y)) :
+    (s.biUnion t).dens = ∑ u ∈ s, (t u).dens := by
+  simp [dens, card_biUnion h, sum_div]
+
+lemma dens_biUnion_le [DecidableEq β] : (s.biUnion t).dens ≤ ∑ a ∈ s, (t a).dens := by
+  simp only [dens, ← sum_div]
+  gcongr
+  · positivity
+  · exact mod_cast card_biUnion_le
+
+lemma dens_eq_sum_dens_fiberwise [DecidableEq α] {f : β → α} {t : Finset β} (h : ∀ x ∈ t, f x ∈ s) :
```

Comment:
Let's phrase this in terms of Set.MapsTo

Reply:
This matches `card_eq_sum_card_fiberwise`